src/lib/parsers/javascript/javascript.ts

```diff
@@ -0,0 +1,89 @@
+import * as eslint from 'eslint';
+import * as espree from 'espree';
+
+import { determineMediaTypeForScript } from '../../utils/content-type';
+import { IAsyncHTMLElement, IElementFound, IFetchEnd, IScriptParse, Parser } from '../../types';
+import { Sonarwhal } from '../../sonarwhal';
+
+const scriptContentRegex: RegExp = /^<script[^>]*>([\s\S]*)<\/script>$/;
+const defaultParserOptions = {
+    comment: true,
+    ecmaVersion: 8,
+    loc: true,
+    range: true,
+    tokens: true
+};
+
+export default class JavascriptParser extends Parser {
+    public constructor(sonarwhal: Sonarwhal) {
+        super(sonarwhal);
+
+        sonarwhal.on('fetch::end', this.parseJavascript.bind(this));
+        sonarwhal.on('element::script', this.parseJavascriptTag.bind(this));
+    }
+
+    private async emitScript(code: string, resource: string) {
+        const ast = espree.parse(code, defaultParserOptions);
+
+        const scriptData: IScriptParse = {
+            resource,
+            sourceCode: new eslint.SourceCode(code, ast)
+        };
+
+        await this.sonarwhal.emitAsync('parse::javascript', scriptData);
+    }
+
+    private parseJavascript(fetchEnd: IFetchEnd) {
+        if (fetchEnd.response.mediaType !== 'text/javascript') {
```

Comment:
What about `application/javascript`? I know a few issues were opened in sonarwhal discussing which media type is right, but here maybe both should be considered?

Reply:
The `response.mediaType` is calculated in sonar, and in case of a javascript, it will be always `text/javascript`